### PR TITLE
Comma operator in JS

### PR DESCRIPTION
**Pull request by bhaskarmelkani**
## TL;DR;

When placed in an expression, it evaluates every expression from left to right and returns the right most.
## Username

 [@bhaskarmelkani](https://www.twitter.com/bhaskarmelkani)

### CODE REVIEW

File: _posts/en/2016-xx-xx-comma-operaton-in-js.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+i:1, j:2
+i:2, j:4
+i:3, j:6
+i:4, j:8
+```
+
+But when placed in an expression, it evaluates every expression from left and returns the right most expression.
+
+Eg:-
+```
```
Comment (loverajoel) [inline _posts/en/2016-xx-xx-comma-operaton-in-js.md]: add ```js

File: _posts/en/2016-xx-xx-comma-operaton-in-js.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+function a(){console.log('a'); return 'a';} 
+function b(){console.log('b'); return 'b';} 
+function c(){console.log('c'); return 'c';}
+
+x = (a(), b(), c());
+
+console.log(x);      // Outputs "c"
+```
+Output:-
+```
```
Comment (loverajoel) [inline _posts/en/2016-xx-xx-comma-operaton-in-js.md]: same here

File: _posts/en/2016-xx-xx-comma-operaton-in-js.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+Apart from being just a delimieter, the comma operator allows you to put multiple statement in a place where one statemenet is expected.
+Eg:-
+```
+for(var i=0, j=0; i<5; i++, j++, j++){
+  console.log("i:"+i+", j:"+j);
+}
+```
+
+Output:-
+```
```
Comment (loverajoel) [inline _posts/en/2016-xx-xx-comma-operaton-in-js.md]: add ```js

File: _posts/en/2016-xx-xx-comma-operaton-in-js.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+tip-username: bhaskarmelkani
+tip-username-profile: https://www.twitter.com/bhaskarmelkani
+tip-tldr: When placed in an expression, it evaluates every expression from left and returns the last one.
+
+categories:
+    - en
+---
+Apart from being just a delimieter, the comma operator allows you to put multiple statement in a place where one statemenet is expected.
+Eg:-
+```
```
Comment (loverajoel) [inline _posts/en/2016-xx-xx-comma-operaton-in-js.md]: add ```js

**Comment by bhaskarmelkani** [comment]
Done :)

File: _posts/en/2016-xx-xx-comma-operaton-in-js.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+title: Comma operator in JS
+tip-number: xx
+tip-username: bhaskarmelkani
+tip-username-profile: https://www.twitter.com/bhaskarmelkani
+tip-tldr: When placed in an expression, it evaluates every expression from left and returns the last one.
+
+categories:
+    - en
+---
+Apart from being just a delimieter, the comma operator allows you to put multiple statement in a place where one statemenet is expected.
```
Comment (zenopopovici) [inline _posts/en/2016-xx-xx-comma-operaton-in-js.md]: Spelling mistakes all over the place. Please correct.

**Comment by bhaskarmelkani** [comment]
Sorry my bad. 
Corrected.
Thanks :)

File: _posts/en/2016-xx-xx-comma-operaton-in-js.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+title: Comma operator in JS
+tip-number: xx
+tip-username: bhaskarmelkani
+tip-username-profile: https://www.twitter.com/bhaskarmelkani
+tip-tldr: When placed in an expression, it evaluates every expression from left and returns the last one.
+
+categories:
+    - en
+---
+Apart from being just a delimiter, the comma operator allows you to put multiple statement in a place where one statement is expected.
```
Comment (zenopopovici) [inline _posts/en/2016-xx-xx-comma-operaton-in-js.md]: `multiple statements`

File: _posts/en/2016-xx-xx-comma-operaton-in-js.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+Output:-
+```js
+i:0, j:0
+i:1, j:2
+i:2, j:4
+i:3, j:6
+i:4, j:8
+```
+
+But when placed in an expression, it evaluates every expression from left and returns the right most expression.
```
Comment (zenopopovici) [inline _posts/en/2016-xx-xx-comma-operaton-in-js.md]: 1. Remove `But`. Start with `When`
2. `from left to right`

File: _posts/en/2016-xx-xx-comma-operaton-in-js.md [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+Output:-
+```js
+"a"
+"b"
+"c"
+
+"c"
+```
+
+* Note: The comma(`,`) operator has the lowest priority of all the javascript operators, so without the parenthesis the expression would become- ` (x = a()), b(), c();  `.
```
Comment (zenopopovici) [inline _posts/en/2016-xx-xx-comma-operaton-in-js.md]: 1. `all javascript operators` (remote `the`).
2. `would become:`

**Comment by bhaskarmelkani** [comment]
Done, thanks for the cooperation.

File: _posts/en/2016-xx-xx-comma-operaton-in-js.md [hunk 9]
```diff
@@ -0,0 +1,51 @@
+---
+layout: post
+
+title: Comma operator in JS
+tip-number: xx
```
Comment (loverajoel) [inline _posts/en/2016-xx-xx-comma-operaton-in-js.md]: please update the number to 57

**Comment by loverajoel** [comment]
- Rename the file to `2016-08-09-comma-operaton-in-js.md`

**Comment by bhaskarmelkani** [comment]
Done :)

**Comment by loverajoel** [comment]
@bhaskarmelkani Hey, now we can embed jsbin playground, can you create some snippet like this for this tip? http://www.jstips.co/en/copy-to-clipboard/

File: _posts/en/2016-08-09-comma-operaton-in-js.md [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+
+When placed in an expression, it evaluates every expression from left to right and returns the right most expression.
+
+Eg:-
+```js
+function a(){console.log('a'); return 'a';} 
+function b(){console.log('b'); return 'b';} 
+function c(){console.log('c'); return 'c';}
+
+x = (a(), b(), c());
```
Comment (loverajoel) [inline _posts/en/2016-08-09-comma-operaton-in-js.md]: `var x`

**Comment by loverajoel** [comment]
@bhaskarmelkani what do you think? https://jsbin.com/vimogap/3/edit?js,console

**Comment by bhaskarmelkani** [comment]
Looks good to me 👍

**Comment by bhaskarmelkani** [comment]
Should I embed https://jsbin.com/vimogap/3/edit?js,console ?

**Comment by bhaskarmelkani** [comment]
@loverajoel

**Comment by loverajoel** [comment]
@bhaskarmelkani you should!

### Now you can be rewarded for your tips

[Support the cominity](https://github.com/loverajoel/jstips#support-the-community)
[How to do that?](https://github.com/loverajoel/jstips/blob/gh-pages/CONTRIBUTING.md#requirements)

**Comment by bhaskarmelkani** [comment]
@loverajoel check if it looks good to you, or it needs any correction

**Comment by loverajoel** [comment]
Thanks!

**Pull request merged**